source/MaterialXGenShader/Shader.h

```diff
@@ -217,7 +241,10 @@ class Shader
         std::queue<Brackets> scopes;
         std::set<string> includes;
         std::set<ShaderImplementation*> definedFunctions;
-        
+
+        // Block holding constant variables for this stage
+        VariableBlockPtr constants;
```

Comment:
Not sure we need a pointer here, and create it dynamically. This could be a block that always exists to simplify the logic, same as for `VariableBlock _appData` on the Shader class.

Reply:
I wasn't sure if there is a need for vertex (or any other future) stage constants at this point but I assume it's not much overhead.